mcprep-build.py

```diff
@@ -0,0 +1,101 @@
+import os
+import shutil
+import sys
+from pathlib import Path
+from typing import List
+
+# Config stuff
+BUILD_NAME: str = "MCprep_addon"
+
+# Internals
+BUILD_DIRECTORY: Path = Path("build")
+INTERMIDIATE_PATH: Path = BUILD_DIRECTORY / Path("inter")
+ADDON_DIRECTORY: Path = Path("MCprep_addon")
+BLENDER_INSTALLS: Path = Path("blender_installs.txt")
+DEBUG_FILE: Path = Path("mcprep_dev.txt")
+BUILT_ZIP: Path = BUILD_DIRECTORY / Path(BUILD_NAME + ".zip")
+
+
+def main():
+    blender_installs: List[Path] = []
+    debug_mode: bool = False
+
+    # Check arguments to see what mode to run the compiler in
+    if len(sys.argv) <= 1:
+        pass
+    elif sys.argv[1] == "-d" or sys.argv[1] == "--debug":
+        debug_mode = True
+    else:
+        print("Not a valid option!")
+        return
+
+    # Check if the developer blender_installs.txt
+    if not BLENDER_INSTALLS.exists():
+        print("You need a blender_installs.txt file!")
+        return
+
+    # If this happens, I've lost all hope for hummanity
+    if BLENDER_INSTALLS.is_dir():
+        print("blender_installs.txt needs to be a text file!")
+        return
+
+    # Read in file paths and return an error if there's no paths
+    with open(BLENDER_INSTALLS, "r") as f:
+        for line in f:
+            blender_installs.append(Path(line.rstrip()))
+    if not len(blender_installs):
+        print("No autopopulation of Blender install paths for now!")
+        return
+
+    # Check if the addon folder exists
+    if not ADDON_DIRECTORY.exists():
+        print("Addon folder does not exist!")
+        return
+
+    # Create build directory so we don't get errors
+    if not BUILD_DIRECTORY.exists():
+        BUILD_DIRECTORY.mkdir()
+
+    # Remove the built zip so we don't get errors
+    if BUILT_ZIP.exists():
+        os.remove(BUILT_ZIP)
+
+    # Create archive and move it to the build directory since shutil makes
+    # the archive in the current working directory
+    shutil.make_archive(BUILD_NAME, "zip", ADDON_DIRECTORY)
```

Comment:
Doesn't this copy all of the git hidden folders and whatnot?

We should probably include an ignore or include list here. More to maintain, but helps enforce consistent building by different individuals (and maybe it's eventually automated on a server while doing releases, all the better to ensure no locality differences).

Reply:
Depends on the addon folder. MCprep's code is in a seperate folder, so only the actual code is copied